core/src/modules/ops/export.ts

```diff
@@ -175,7 +175,7 @@ export namespace ExportOps {
     saveExports = true
   ) {
     const where: WhereOptions<Export> = {
-      state: "failed",
+      state: { [Op.in]: ["failed", "canceled"] },
```

Comment:
Yeah, this could affect the existing "retry failed" functionality. After #2818, we are now canceling older exports if there are multiple for the same record/destination ([test case here](https://github.com/grouparoo/grouparoo/blob/eb40f33317cb9a003676f2bb74d358f55a15a9d8/core/__tests__/models/destination/plugins/exportRecords.ts#L695)). This means that these intentionally canceled ones would also be retried when doing it by date range.

Seeing this and the date range thing makes me thing that perhaps this is a different method, `retryExportById` or similar. To ensure this new one and the existing one are retrying in the same way we could break out the actual updating of the export into a separate function and share that.